### ISSUE TITLE
compile warnings with gcc 11.2 in brotli

### ISSUE THREAD

**Issue by GitMensch** [rr-debugger/rr#3232]
~~~
[  1%] Building C object CMakeFiles/brotli.dir/third-party/brotli/dec/decode.c.o
/home/install/rr/third-party/brotli/dec/decode.c:1954:41: warning: argument 2 of type ‘const uint8_t *’ {aka ‘const unsigned char *’} declared as a pointer [-Wvla-parameter]
 1954 |     size_t encoded_size, const uint8_t* encoded_buffer, size_t* decoded_size,
      |                          ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
In file included from /home/install/rr/third-party/brotli/dec/decode.c:7:
/home/install/rr/third-party/brotli/include/brotli/decode.h:204:19: note: previously declared as a variable length array ‘const uint8_t[*decoded_size]’ {aka ‘const unsigned char[*decoded_size]’}
  204 |     const uint8_t encoded_buffer[BROTLI_ARRAY_PARAM(encoded_size)],
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/install/rr/third-party/brotli/dec/decode.c:1955:14: warning: argument 4 of type ‘uint8_t *’ {aka ‘unsigned char *’} declared as a pointer [-Wvla-parameter]
 1955 |     uint8_t* decoded_buffer) {
      |     ~~~~~~~~~^~~~~~~~~~~~~~
In file included from /home/install/rr/third-party/brotli/dec/decode.c:7:
/home/install/rr/third-party/brotli/include/brotli/decode.h:206:13: note: previously declared as a variable length array ‘uint8_t[encoded_size]’ {aka ‘unsigned char[encoded_size]’}
  206 |     uint8_t decoded_buffer[BROTLI_ARRAY_PARAM(*decoded_size)]);
      |     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[  2%] Building C object CMakeFiles/brotli.dir/third-party/brotli/enc/encode.c.o
/home/install/rr/third-party/brotli/enc/encode.c:1433:20: warning: argument 5 of type ‘const uint8_t *’ {aka ‘const unsigned char *’} declared as a pointer [-Wvla-parameter]
 1433 |     const uint8_t* input_buffer, size_t* encoded_size,
      |     ~~~~~~~~~~~~~~~^~~~~~~~~~~~
In file included from /home/install/rr/third-party/brotli/enc/encode.c:9:
/home/install/rr/third-party/brotli/include/brotli/encode.h:293:19: note: previously declared as a variable length array ‘const uint8_t[*encoded_size]’ {aka ‘const unsigned char[*encoded_size]’}
  293 |     const uint8_t input_buffer[BROTLI_ARRAY_PARAM(input_size)],
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/install/rr/third-party/brotli/enc/encode.c:1434:14: warning: argument 7 of type ‘uint8_t *’ {aka ‘unsigned char *’} declared as a pointer [-Wvla-parameter]
 1434 |     uint8_t* encoded_buffer) {
      |     ~~~~~~~~~^~~~~~~~~~~~~~
In file included from /home/install/rr/third-party/brotli/enc/encode.c:9:
/home/install/rr/third-party/brotli/include/brotli/encode.h:295:13: note: previously declared as a variable length array ‘uint8_t[input_size]’ {aka ‘unsigned char[input_size]’}
  295 |     uint8_t encoded_buffer[BROTLI_ARRAY_PARAM(*encoded_size)]);
      |     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~

That happens on a fresh installed Ubuntu.

**Comment by GitMensch** [comment]
Is there an active upstream to check / ask?

**Comment by rocallahan** [comment]
There should be

**Comment by GitMensch** [comment]
Those are fixed upstream, I'm checking for a new release, if not available I'd create a pull-request with updating it to the fixed version (which then has no release number, so would be 1.0.9.32 (which is 1.0.9, itself a re-released 1.0.8 + 32 commits).

**Comment by rocallahan** [comment]
We want to keep copying it in the same way it is now. You can reference the Google Github commit the same way we currently do.
